pkg/common/consumer/consumer_factory.go

```diff
@@ -64,6 +63,7 @@ func (c kafkaConsumerGroupFactoryImpl) StartConsumerGroup(groupID string, topics
 	ctx, cancel := context.WithCancel(context.Background())
 
 	go func() {
+		defer close(errorCh)
```

Comment:
While you are at it, what about moving `errorCh := make(chan error, 10)` inside the lambda? Clearer lifecycle.